app/src/main/java/com/github/openwebnet/repository/impl/EnvironmentRepositoryImpl.java

```diff
@@ -102,4 +106,19 @@ public EnvironmentRepositoryImpl() {
         });
     }
 
+    @Override
+    public Observable<Void> delete(Integer id) {
+        return Observable.create(subscriber -> {
+            try {
```

Comment:
If we wanna play with reflections on android you should have a look at [this](https://github.com/niqdev/enhanced-domotic/tree/master/src/main/java/com/domotic/enhanced/reflection)